mesonbuild/backend/ninjabackend.py

```diff
@@ -2002,10 +2005,13 @@ def generate_rust_target(self, target: build.BuildTarget) -> None:
             args += ['-L', d]
         has_shared_deps = any(isinstance(dep, build.SharedLibrary) for dep in target.get_dependencies())
         if isinstance(target, build.SharedLibrary) or has_shared_deps:
-            # add prefer-dynamic if any of the Rust libraries we link
-            # against are dynamic, otherwise we'll end up with
-            # multiple implementations of crates
-            args += ['-C', 'prefer-dynamic']
+            has_rust_shared_deps = any(isinstance(dep, build.SharedLibrary) and dep.uses_rust() and dep.rust_crate_type != 'cdylib'
+                                       for dep in target.get_dependencies())
```

Comment:
We call `target.get_dependencies()` already a few lines above, would be nice to reuse the result. Implementation does a recursive traversal of all deps which can take time. Ideally we should profile and use `@lru_cache` where needed, but seems easy to at least not make it worse here for now.

Reply:
Other than that, LGTM

Reply:
oh, I was just suggesting doing `target_deps = target.get_dependencies()` then iterate that list twice, that would be good enough IMHO.

Reply:
Ah, sure. Changed it like that